### PR TITLE
ci: add zizmor for github actions security

### PR DESCRIPTION
**Pull request by Sysix**
> zizmor is a static analysis tool for GitHub Actions. It can find many common security issues in typical GitHub Actions CI/CD setups.

https://woodruffw.github.io/zizmor/

EDIT: what is the right PR-syntax for this?

### CODE REVIEW

File: .github/workflows/prepare_release_oxlint.yml [hunk 1]
```diff
@@ -10,11 +10,6 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
-permissions:
```
Comment (Sysix) [inline .github/workflows/prepare_release_oxlint.yml:13]: I hope I did split them right and did not missed one on a job

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: Check PR
 
 on:
-  pull_request_target:
```
Comment (Sysix) [inline .github/workflows/pr.yml:4]: https://woodruffw.github.io/zizmor/audits/#dangerous-triggers

Comment (Boshen) [reply .github/workflows/pr.yml:4]: https://github.com/amannn/action-semantic-pull-request?tab=readme-ov-file#event-triggers

Comment (Boshen) [reply .github/workflows/pr.yml:4]: I'm in favor of removing of this check with something else, but not right now.

**Review by Boshen — APPROVED** [review]
Thank you for this, I've always wanted to harden the security, but the only thing I did was changing tokens to another account with less privileges.

**Pull request merged**